File: batches_java/batch_06_loop_type_change/compare.java
public class SumList {
    public static int sumList(int[] nums) {
        int total = 0;
        for (int n : nums) {
            total += n;
        }
        return total;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        System.out.println(sumList(nums));
    }
}

File: batches/batches/batch_06_loop_type_change/compare.py
"""
Comparison - Loop Type Change for↔while (NOT cheating)
Same logic, for-loop rewritten as while-loop.
"""


def sum_list(nums):
    total = 0
    i = 0
    while i < len(nums):
        total += nums[i]
        i += 1
    return total


if __name__ == "__main__":
    print(sum_list([1, 2, 3, 4, 5]))
